lib/rules/no-global-setup.js

```diff
@@ -10,24 +10,37 @@ var setupRegexp = /^(before|after)(Each|All)$/
 
 module.exports = function (context) {
   var suiteDepth = 0
+  var hasGlobalSetup = false
+  var hasRootSuite = false
+  var globalSetupNames = []
   return {
     CallExpression: function (node) {
       if (suiteRegexp.test(node.callee.name)) {
+        hasRootSuite = true
         suiteDepth++
       } else if (setupRegexp.test(node.callee.name) && suiteDepth === 0) {
-        context.report({
-          data: {
-            name: node.callee.name
-          },
-          message: 'Do not use `{{name}}` outside a `describe`.',
-          node
-        })
+        hasGlobalSetup = true
+        globalSetupNames.push(node.callee.name)
       }
     },
     'CallExpression:exit': function (node) {
       if (suiteRegexp.test(node.callee.name)) {
         suiteDepth--
       }
+    },
+    'Program:exit': function () {
+      if (hasGlobalSetup && hasRootSuite) {
+        context.report({
+          data: {
+            names: globalSetupNames.join(', ')
+          },
+          message: 'Do not use `{{names}}` outside a `describe` except for in global helpers.',
+          loc: {
```

Comment:
The challenge here is that the cleanest way to do this is to track the parsing of all nodes down-and-up the AST back to the root `Program` level... Then evaluate the global condition on the `Program:exit`.

So, the evaluation unit doesn't know if the exception-condition (e.g. "No `describe` block in the unit's global context when there is one-or-more global setup blocks") has been met while inside a node.

Going to try collecting the nodes in a globaSetupNodes module collection and then traversing that collection and emitting a context.report for each node post-AST traversal.  Gimme a bit... ;-)